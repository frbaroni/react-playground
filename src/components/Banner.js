import React, { Component } from "react";
import { Carousel, CarouselItem } from "reactstrap";

import img0 from "../slider/a.jpg";
import img1 from "../slider/b.jpg";
import logo from "../slider/logo.jpg";
const items = [img0, img1];

class Banner extends Component {
  constructor(props) {
    super(props);
    this.state = { activeIndex: 0 };
  }

  onEntering = (node, isAppearing) => {
    // console.log("Enter", node);
  };

  onExited = node => {
    // console.log("Exit", node);
  };

  next = () => {
    const nextIndex =
      this.state.activeIndex === items.length - 1
        ? 0
        : this.state.activeIndex + 1;
    this.setState({ activeIndex: nextIndex });
  };

  previous = () => {
    this.next();
  };

  slides = () => {
    return items.map(item => (
      <CarouselItem
        onEntering={this.onEntering}
        onEnter={this.onEntering}
        onEntered={this.onEntering}
        onExited={this.onExited}
        key={item}
        src={item}
        altText={item}
      />
    ));
  };

  logo = () => {
    return (
      <div className="logo">
        <a href="#logo">
          <img src={logo} alt="Hello World" />
        </a>
      </div>
    );
  };

  render() {
    return (
      <div className="col-sm-12 pull-sm-12">
        <Carousel
          activeIndex={this.state.activeIndex}
          next={this.next}
          previous={this.previous}
          pause={false}
          slide={false}
          interval={2000}
          keyboard={false}
          ride={"carousel"}
        >
          {this.slides()}
        </Carousel>
        {this.logo()}
      </div>
    );
  }
}

export default Banner;
