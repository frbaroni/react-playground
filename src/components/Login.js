import React, { Component } from "react";
import PropTypes from "prop-types";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: null,
      password: null,
      rememberMe: false
    };
  }

  update = name => {
    return e => {
      const t = e.target;
      this.setState({
        [name]: t.type === "checkbox" ? t.checked : t.value
      });
    };
  };
  onEmail = this.update("email");
  onPassword = this.update("password");
  onRememberMe = this.update("rememberMe");
  onSubmit = e => {
    console.log("Do Login", this.state);
  };

  mayRenderError = () => {
    if (!this.props.error) {
      return false;
    }
    return (
      <div className="alert alert-danger">
        <strong>Error!</strong> invalid credentials.
      </div>
    );
  };

  render() {
    return (
      <div className="form col-md-6 offset-md-3 col-sm-12 push-sm-12">
        <form>
          <div className="form-group">
            <label htmlFor="exampleInputEmail1">Email address</label>
            <input
              type="email"
              className="form-control"
              placeholder="Enter email"
              onChange={this.onEmail}
            />
          </div>
          <div className="form-group">
            <label htmlFor="exampleInputPassword1">Password</label>
            <input
              type="password"
              className="form-control"
              placeholder="Password"
              onChange={this.onPassword}
            />
          </div>
          <div className="form-check">
            <label className="form-check-label">
              <input
                type="checkbox"
                className="form-check-input"
                onChange={this.onRememberMe}
              />
              Remember me?
            </label>
          </div>
          <button
            type="button"
            className="btn btn-primary"
            onClick={this.onSubmit}
          >
            Login
          </button>
          {this.mayRenderError()}
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  error: PropTypes.bool.isRequired
};

export default Login;
