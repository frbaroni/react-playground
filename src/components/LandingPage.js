import React from "react";

import Banner from "./Banner";
import Login from "./LoginContainer";

class LandingPage extends React.Component {
  render() {
    return (
      <div className="row wrapper">
        <Banner />
        <Login />
      </div>
    );
  }
}

export default LandingPage;
