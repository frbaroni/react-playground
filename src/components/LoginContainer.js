import { connect } from "react-redux";
import { login } from "../actions/authActions";
import { Login } from "./Login";

function mapStateToProps(state) {
  return {
    error: state.auth.error,
    success: state.auth.success
  };
}

const mapDispatchToProps = {
  login
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
