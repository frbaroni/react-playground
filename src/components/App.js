import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";

import React, { Component } from "react";
import playgroundApp from "../reducers";
import LandingPage from "./LandingPage";

let store = createStore(
  playgroundApp,
  composeWithDevTools(applyMiddleware(thunk))
);

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <LandingPage />
      </Provider>
    );
  }
}

export default App;
