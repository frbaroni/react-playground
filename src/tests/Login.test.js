import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { Login } from "../components/Login";

describe("Login component", () => {
  it("renders pristine", () => {
    const props = {
      success: false,
      error: false,
      login: jest.fn()
    };
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with error", () => {
    const props = {
      success: false,
      error: true,
      login: jest.fn()
    };
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with success", () => {
    const props = {
      success: true,
      error: false,
      login: jest.fn()
    };
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders with success and error", () => {
    const props = {
      success: true,
      error: true,
      login: jest.fn()
    };
    const wrapper = shallow(<Login {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
