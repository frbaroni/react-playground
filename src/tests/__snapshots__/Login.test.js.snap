// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders pristine 1`] = `
<div
  className="form col-md-6 offset-md-3 col-sm-12 push-sm-12"
>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </label>
    </div>
    <div
      className="form-check"
    >
      <label
        className="form-check-label"
      >
        <input
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        Remember me?
      </label>
    </div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`Login component renders with error 1`] = `
<div
  className="form col-md-6 offset-md-3 col-sm-12 push-sm-12"
>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </label>
    </div>
    <div
      className="form-check"
    >
      <label
        className="form-check-label"
      >
        <input
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        Remember me?
      </label>
    </div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <div
      className="alert alert-danger"
    >
      <strong>
        Error!
      </strong>
       invalid credentials.
    </div>
  </form>
</div>
`;

exports[`Login component renders with success 1`] = `
<div
  className="form col-md-6 offset-md-3 col-sm-12 push-sm-12"
>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </label>
    </div>
    <div
      className="form-check"
    >
      <label
        className="form-check-label"
      >
        <input
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        Remember me?
      </label>
    </div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <div
      className="alert alert-success"
    >
      <strong>
        Success!
      </strong>
       you're being redirected.
    </div>
  </form>
</div>
`;

exports[`Login component renders with success and error 1`] = `
<div
  className="form col-md-6 offset-md-3 col-sm-12 push-sm-12"
>
  <form>
    <div
      className="form-group"
    >
      <label>
        Email address
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Email address"
          type="email"
        />
      </label>
    </div>
    <div
      className="form-group"
    >
      <label>
        Password
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </label>
    </div>
    <div
      className="form-check"
    >
      <label
        className="form-check-label"
      >
        <input
          className="form-check-input"
          onChange={[Function]}
          type="checkbox"
        />
        Remember me?
      </label>
    </div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <div
      className="alert alert-danger"
    >
      <strong>
        Error!
      </strong>
       invalid credentials.
    </div>
    <div
      className="alert alert-success"
    >
      <strong>
        Success!
      </strong>
       you're being redirected.
    </div>
  </form>
</div>
`;
